src/runtimes/docker/utils/getNativeArchitecture.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+
+/**
+ * Returns native architecture of the current machine
+ * @returns native architecture of the current machine
+ */
+export function getNativeArchitecture(): 'amd64' | '386' | 'arm64' | 'arm' {
```

Comment:
Also, consider moving this helper into utils/osUtils.ts.

Reply:
It might be cleaner to have an explicit type for the possible return values (`type ContainerArchitectures = 'amd64' | '386' | ...`) instead of just having them defined on the return value. Makes it easier to re-use them if we need to.

Reply:
The neat syntax we have in `src/utils/platform.ts` might be helpful for this as well.

Reply:
(Recapping our discussion offline) That syntax is still neat but we won't need it here; we don't have a scenario for iterating through the list of OSs / architectures.